spec/forms/providers/proceeding_merits_task/vary_order_form_spec.rb

```diff
@@ -0,0 +1,25 @@
+require "rails_helper"
+
+RSpec.describe Providers::ProceedingMeritsTask::VaryOrderForm do
+  subject(:form) { described_class.new(params) }
+
+  let(:params) { { details: } }
+  let(:details) { Faker::Lorem.sentence }
```

Comment:
This has been mentioned before, but I think it's important we write deterministic tests.

If we introduce randomness, we invite flakiness —and flaky tests are useless tests.

If we truly are concerned about exploring the problem space, and ensuring our form can handle all manner of different inputs, then we should be specific and explicit —and write deterministic tests that cover those cases.

I suspect we aren't though, and that some arbitrary test data would suffice.

What do you think?

[This article by Martin Fowler](https://martinfowler.com/articles/nonDeterminism.html) puts it much better than I ever could, if people are interested.

Reply:
i agree, mostly, but in terms of free text i think non-deterministic can cover the random nature of inputs and is a pattern I see elsewhere and have used before. I have found non-deterministic tests which can be fail when it randomly puts characters that one would not expect and these can expose problems that might otherwise only be caught by "live" testing.

In this instance I am therefore not inclined to remove but do not feel particularly strongly about it

Reply:
My point is that a unit test that sometimes fails and sometimes passes is useless, because it can give us no confidence in our system.

What are we trying to test?

That the form can handle atypical characters? 
(e.g `I can input \";* strange and unexpected characters`)

Or that it is safe from XSS vulnerabilities? 
(e.g `Don't click this <script>alert("This is really bad!");</script>`)

Or that is can handle specific random combinations of words? 
(e.g `sausages and bacon breaks the form`)

As I said, if we are really interested in exploring the problem space and testing a range of inputs, then we should be explicit about them and write deterministic tests to cover them.

Reply:
Clearly you feel strongly about it, but personally I don't think we need to sweat too much over such things.

I will remove